dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java

```diff
@@ -1131,6 +1135,19 @@ private boolean checkAndImport(User loginUser, long projectCode, Map<String, Obj
             }
             taskDefinitionLogList.add(taskDefinitionLog);
         }
+        //fix taskParams when taskType is SWITCH
+        for (TaskDefinitionLog taskDefinitionLog: taskDefinitionLogList) {
+            if (TaskType.SWITCH.getDesc().equals(taskDefinitionLog.getTaskType())) {
+                String taskParams = taskDefinitionLog.getTaskParams();
+                Matcher matcher = SWITCH_NODE_REGEX.matcher(taskParams);
+                while (matcher.find()) {
+                    String oldTaskCode = matcher.group(1).trim();
+                    Long newTaskCode = taskCodeMap.get(Long.parseLong(oldTaskCode));
+                    taskParams = taskParams.replaceAll(oldTaskCode, newTaskCode == null ? oldTaskCode : newTaskCode.toString());
+                }
+                taskDefinitionLog.setTaskParams(taskParams);
```

Comment:
This code should be moved into the section above lines 1117-1137 dealing with taskDefinitionList